src/main/scala/firrtl/passes/Inline.scala

```diff
@@ -0,0 +1,171 @@
+package firrtl
+package passes
+
+// Datastructures
+import scala.collection.mutable
+
+import firrtl.Mappers.{ExpMap,StmtMap}
+import firrtl.Utils.WithAs
+
+
+// Tags an annotation to be consumed by this pass
+case object InlineCAKind extends CircuitAnnotationKind
+
+// Only use on legal Firrtl. Specifically, the restriction of
+//  instance loops must have been checked, or else this pass can
+//  infinitely recurse
+object InlineInstances extends Transform {
+   val inlineDelim = "$"
+   def name = "Inline Instances"
+   def execute(circuit: Circuit, annotations: Seq[CircuitAnnotation]): TransformResult = {
```

Comment:
This isn't going to work just using "$" as a delim, as with the previous lowering issues, this can lead to collisions (try adding output i$a to any of the InlineInstancetests). I can look in to modifying this pass to use the (currently private) utility functions in Uniquify to achieve the same affect. If I do that, we can also use _ as the delimiter instead.